docs/features/software-templates/experimental.md

```diff
@@ -13,7 +13,64 @@ Please leave feedback on these features in the [Backstage Discord](https://disco
 
 ## Retries and Recovery
 
-### TODO
+Do you have long-running tasks, and can they recover and proceed after Backstage redeploy?
+
+Then you definitely will benefit from enabling the experimental scaffolder task recovery feature.
+
+Whenever you do redeploy, on startup there will be a check of all tasks in "processing" state that you identified in your template as they are capable of starting over.
+
+This is an example of how you can do it:
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: recoverable-template
+spec:
+  EXPERIMENTAL_recovery:
+    EXPERIMENTAL_strategy: startOver
+```
+
+And to enable the recovery feature itself, you can do it by adding this snippet into your app-config.yaml file:
+
+```yaml
+scaffolder:
+  EXPERIMENTAL_recoverTasks: true
+```
+
+By default, all the tasks that are in a processing state and have a heartbeat of more than 30 seconds will be automatically recovered.
+This implies that the task's status will shift to "Open," initiating its execution from the beginning. So you have to be sure that your tasks can run multiple times.
```

Comment:
Could you help me understand this part a little more? Tasks will shift to an Open status if they where in processing and had a heartbeat of more than 30 seconds post redeploy/interruption? and then re-initialise execution? 💭

Reply:
Maybe I have to rephrase it if it's confusing 😅 

The gist is that after Backstage restart all tasks which were running would be in "processing" state. 
Though they will be doing nothing.

The default behaviour now in Backstage to change the status to "failed" when certain timeout of no heartbeat reached. 

In case when you want to make the process of running tasks seamless to users (no failed tasks due to Backstage restart), and not add extra work of checking that no running tasks now in the queue, so Backstage can be restarted, the recovery feature can be enabled.

When such task "in limbo" (processing state, no heartbeat for a certain time) is found, we will start executing the task from the start (that what happens if to change the status of task to "Open").

Reply:
Oh I see, I think if we explain this behaviour in that context it will be easier to understand 👍  WDYT? Like a TLDR version 😅

Reply:
Included the visualization of it, usually it's easier to grasp :)

Reply:
@benjdlambert Is it ok to add visuals to docs or better to avoid it?

Reply:
I think we can skip for now, I reworded some things and hopefully it's little better.

Thinking that we can introduce diagrams later, but would rather do it with draw.io or whatever we've used in other places so that they can be edited over time rather than a png 👍